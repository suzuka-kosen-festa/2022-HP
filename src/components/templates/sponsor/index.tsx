import type { FC } from "react";
import tw from "twin.macro";
import { Link } from "wouter";
import type { SponsorProperties } from "./types/model";
import { Title } from "../../molecules/title";
import { Text } from "../../atoms/text";
import { Button } from "../../atoms/button";

const Main = tw.main`pt-1 pb-8 sm:(pt-6 pb-40)`;

const SponsorBox = tw.div`flex flex-col items-center`;

const Sponsor: FC<SponsorProperties> = ({ sponsors }) => (
  <Main aria-label="Sponsor page">
    <Title css={tw`mb-5.5 sm:mb-14`} colorKey="red">
      Sponsor
    </Title>
    <SponsorBox>
      <h2 css={tw`font-zen font-bold text-text text-base mb-8 sm:(text-2xl mb-20)`}>
        第57回鈴鹿高専祭を開催するにあたり、支援してくださった企業様のお名前を以下に記載させていただきます。
      </h2>
      <ul css={tw`mb-8 sm:mb-40`}>
        {sponsors.map(sponsor => (
          <li key={sponsor.id}>
            <Text css={tw`text-base sm:text-2xl after:content-[" 様"]`}>{sponsor.name}</Text>
          </li>
        ))}
      </ul>
      <Link to="/">
        <Button textStyles={tw`text-white`} colorKey="red" as="a" role="button">
          トップページに戻る
        </Button>
      </Link>
    </SponsorBox>
  </Main>
);

export { Sponsor };
