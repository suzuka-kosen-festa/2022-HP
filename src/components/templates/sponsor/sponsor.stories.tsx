import type { ComponentStoryObj, ComponentMeta } from "@storybook/react";
import { ComponentPropsWithoutRef } from "react";
import { Sponsor } from ".";

type T = typeof Sponsor;
type Story = ComponentStoryObj<T>;
type Meta = ComponentMeta<T>;

const args: ComponentPropsWithoutRef<T> = {
  sponsors: [
    {
      name: "株式会社 鈴鹿",
      id: 1,
    },
    {
      name: "有限会社 高専",
      id: 2,
    },
    {
      name: "〇〇クリニック",
      id: 3,
    },
  ],
};

export default {
  args,
  argTypes: {
    sponsors: {
      description: "Data list of sponsor",
      control: {
        type: "array",
      },
    },
  },
  component: Sponsor,
} as Meta;

export const Default: Story = {};
