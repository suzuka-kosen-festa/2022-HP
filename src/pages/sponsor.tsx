import type { FC } from "react";
import useSWR from "swr";
import { Redirect } from "wouter";
import { Sponsor } from "../components/templates/sponsor";
import { fetcher } from "../libs/axios";
import type { SponsorCompany } from "../api/prisma";

const SponsorPage: FC = () => {
  const { data, error } = useSWR<ReadonlyArray<SponsorCompany>>("/sponsorcompany", fetcher);
  if (error) return <Redirect to="/error" />;
  if (!data) return <p>Loading...</p>;
  return <Sponsor sponsors={data} />;
};

export { SponsorPage };
