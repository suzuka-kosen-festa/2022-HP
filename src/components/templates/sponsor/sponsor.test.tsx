import { composeStories } from "@storybook/testing-react";
import "@testing-library/jest-dom";
import { render } from "@testing-library/react";
import * as stories from "./sponsor.stories";

const { Default } = composeStories(stories);

const options = {
  name: "Sponsor page",
};

describe("(components) templates/sponsor", () => {
  test("to be templates", () => {
    const { container } = render(<Default />);
    expect(container).toBeTemplate();
  });
  test("to be [role=main]", () => {
    const { getByRole } = render(<Default />);
    expect(getByRole("main", options)).toBeInTheDocument();
  });
  test("take snap shot", () => {
    const { container } = render(<Default />);
    expect(container).toMatchSnapshot();
  });
});
