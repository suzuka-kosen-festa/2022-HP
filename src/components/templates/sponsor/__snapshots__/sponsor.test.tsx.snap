// Vitest Snapshot v1

exports[`(components) templates/sponsor > take snap shot 1`] = `
<div>
  <main
    aria-label="Sponsor page"
    class="css-10jk6q1"
  >
    <div
      class="css-159ot27-Sponsor"
    >
      <img
        alt="ink"
        class="css-mhf7w5-Title"
        height="580"
        loading="eager"
        src="/statics/Ink3_red.webp"
        width="592"
      />
      <h1
        class="css-1ueto1"
      >
        Sponsor
      </h1>
    </div>
    <div
      class="css-dvxtzn"
    >
      <h2
        class="css-1mrebfa-Sponsor"
      >
        第57回鈴鹿高専祭を開催するにあたり、支援してくださった企業様のお名前を以下に記載させていただきます。
      </h2>
      <ul
        class="css-onaqla-Sponsor"
      >
        <li>
          <p
            class="css-13o8352-Sponsor"
          >
            株式会社 鈴鹿
          </p>
        </li>
        <li>
          <p
            class="css-13o8352-Sponsor"
          >
            有限会社 高専
          </p>
        </li>
        <li>
          <p
            class="css-13o8352-Sponsor"
          >
            〇〇クリニック
          </p>
        </li>
      </ul>
      <a
        class="css-voqv4l-Button"
        href="/"
        role="button"
        tabindex="0"
      >
        <p
          class="css-48eax3-Button"
        >
          トップページに戻る
        </p>
      </a>
    </div>
  </main>
</div>
`;
